components/datetime/src/provider/time_zones.rs

```diff
@@ -252,9 +232,11 @@ impl<'a> zerovec::maps::ZeroMapKV<'a> for MetazoneId {
 #[cfg_attr(feature = "datagen", databake(path = icu_datetime::provider::time_zones))]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
 #[yoke(prove_covariance_manually)]
-pub struct MetazonePeriodV1<'data>(
+pub struct MetazonePeriodV1<'data> {
     /// The default mapping between period and metazone id. The second level key is a wall-clock time represented as
     /// the number of minutes since the local [`EPOCH`](icu_timezone::provider::EPOCH). It represents when the metazone started to be used.
     #[cfg_attr(feature = "serde", serde(borrow))]
-    pub ZeroMap2d<'data, TimeZoneBcp47Id, IsoMinutesSinceEpoch, Option<MetazoneId>>,
-);
+    pub list: ZeroMap2d<'data, TimeZoneBcp47Id, IsoMinutesSinceEpoch, NichedOption<MetazoneId, 1>>,
+    /// An XxHash64 checksum of the full metazone names.
```

Comment:
Thought: this increases stack size which is a bit sad, but ZM2D is already big on stack. Should consider fixing in the future.

Reply:
Hmm, it would be nice if we could return things like checksums in the `DataResponseMetadata`, they only need to be checked once at load time, and are `Copy`.

Reply:
Also, checksums are marker-specific, not payload specific. I will try to do this in a followup